src/banking_stage.rs

```diff
@@ -241,63 +239,163 @@ impl Service for BankingStage {
     }
 }
 
-// TODO: When banking is pulled out of RequestStage, add this test back in.
-
-//use bank::Bank;
-//use entry::Entry;
-//use hash::Hash;
-//use record_stage::RecordStage;
-//use record_stage::Signal;
-//use result::Result;
-//use std::sync::mpsc::{channel, Sender};
-//use std::sync::{Arc, Mutex};
-//use std::time::Duration;
-//use transaction::Transaction;
-//
-//#[cfg(test)]
-//mod tests {
-//    use bank::Bank;
-//    use mint::Mint;
-//    use signature::{KeyPair, KeyPairUtil};
-//    use transaction::Transaction;
-//
-//    #[test]
-//    // TODO: Move this test banking_stage. Calling process_transactions() directly
-//    // defeats the purpose of this test.
-//    fn test_banking_sequential_consistency() {
-//        // In this attack we'll demonstrate that a verifier can interpret the ledger
-//        // differently if either the server doesn't signal the ledger to add an
-//        // Entry OR if the verifier tries to parallelize across multiple Entries.
-//        let mint = Mint::new(2);
-//        let bank = Bank::new(&mint);
-//        let banking_stage = EventProcessor::new(bank, &mint.last_id(), None);
-//
-//        // Process a batch that includes a transaction that receives two tokens.
-//        let alice = KeyPair::new();
-//        let tx = Transaction::new(&mint.keypair(), alice.pubkey(), 2, mint.last_id());
-//        let transactions = vec![tx];
-//        let entry0 = banking_stage.process_transactions(transactions).unwrap();
-//
-//        // Process a second batch that spends one of those tokens.
-//        let tx = Transaction::new(&alice, mint.pubkey(), 1, mint.last_id());
-//        let transactions = vec![tx];
-//        let entry1 = banking_stage.process_transactions(transactions).unwrap();
-//
-//        // Collect the ledger and feed it to a new bank.
-//        let entries = vec![entry0, entry1];
-//
-//        // Assert the user holds one token, not two. If the server only output one
-//        // entry, then the second transaction will be rejected, because it drives
-//        // the account balance below zero before the credit is added.
-//        let bank = Bank::new(&mint);
-//        for entry in entries {
-//            assert!(
-//                bank
-//                    .process_transactions(entry.transactions)
-//                    .into_iter()
-//                    .all(|x| x.is_ok())
-//            );
-//        }
-//        assert_eq!(bank.get_balance(&alice.pubkey()), Some(1));
-//    }
-//}
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use bank::Bank;
+    use ledger::Block;
+    use mint::Mint;
+    use packet::{to_packets, PacketRecycler};
+    use signature::{Keypair, KeypairUtil};
+    use std::thread::sleep;
+    use transaction::Transaction;
+
+    #[test]
+    fn test_banking_stage_shutdown() {
+        use logger;
+        logger::setup();
```

Comment:
Were you making use of the logger in any of these tests?

Reply:
did use it, took it out once comfy with paths, will remove